### PR TITLE
Migrate to karma from mocha

### PR DESCRIPTION
**Pull request by clbond**

### CODE REVIEW

**Comment by clbond** [comment]
@SethDavenport

File: package.json [hunk 1]
```diff
@@ -20,19 +19,8 @@
...
-    "exclude": [
-      "**/*.test.js",
-      "test/entry.js"
-    ],
-    "lines": 80,
-    "statements": 80,
-    "branches": 80,
-    "functions": 80
+    "test": "karma start --singleRun true",
+    "test:watch": "karma start --singleRun false"
```
Comment (SethDavenport) [inline package.json]: So there's no longer a coverage threshold? That's OK - we can add it back when we address https://github.com/rangle/rangle-starter/issues/8

File: package.json [hunk 2]
```diff
@@ -108,7 +107,6 @@
     "require-hacker": "^2.1.2",
     "rimraf": "^2.4.3",
     "sinon": "^1.17.3",
```
Comment (SethDavenport) [inline package.json]: sinon is dedundant with jasmine

Comment (SethDavenport) [inline package.json]: chai too

Comment (clbond) [inline package.json]: Alright, I will remove sinon and chai today

File: src/components/form/index.test.js [hunk 3]
```diff
@@ -26,14 +25,19 @@ describe('Form', () => {
...
-    global.document = { activeElement: { blur: onBlur } };
-    const wrapper = shallow(
-      <Form handleSubmit={() => {}} />
-    );
-    const eventStub = { preventDefault: () => {} };
-    wrapper.find('form').simulate('submit', eventStub);
-    assert.isTrue(onBlur.calledOnce, 'focused element was not blurred');
-  });
+  // NOTE(cbond): This test is wild unsafe ... it attempts to monkey-patch
+  // window.document(!) and then does not restore it afterward, raising the
```
Comment (SethDavenport) [inline src/components/form/index.test.js]: yes - can you log an issue for this in rangle-starter?

File: webpack.config.js [hunk 4]
```diff
@@ -22,27 +27,36 @@ module.exports = {
     chunkFilename: '[id].chunk.js',
   },
 
-  devtool: 'source-map',
+  devtool: process.env.NODE_ENV === 'production' ?
+    'source-map' :
```
Comment (SethDavenport) [inline webpack.config.js]: `!devmode` instead?

File: package.json (hunk 1 above)
Comment (clbond) [inline package.json]: Yessir, that's correct. I will add it back, it's very easy: istanbul check-coverage --statement N --branch N --function N

**Comment by SethDavenport** [comment]
As long as it works on IE9 (subject to known CSS issue https://github.com/rangle/rangle-starter/issues/64), :shipit:

**Comment by clbond** [comment]
@SethDavenport Alright, removed sinon and chai dependencies and converted all assertions / spies to Jasmine syntax, and reintroduced coverage thresholds

**Comment by clbond** [comment]
@SethDavenport Good call - my changes broke the project on IE9. (Actually it caused a complete IE crash so it was difficult to debug what the issue was). But changing back to es5-shim from babel-polyfill appears to fix the issue. It runs successfully on IE9 now.

**Pull request merged**